### ISSUE TITLE
Hydrogen Bonding Analysis

### ISSUE THREAD

**Issue by ALescoulie** [Becksteinlab/MDPOW#197]
# mdpow.analysis.hbond

The final analysis I would like to implement is quantifying hydrogen bonding between the solute and solvent. I figure this shouldn't be too difficult in MDAnalysis. This will round out the initial three analyses built using the `EnsembleAnalsis` framework.

**Comment by orbeckst** [comment]
This might need a rewrite of EnsembleAnalysis to directly use a full analysis over a Universe. If so, first raise a separate issue for those changes to the framework and then come back here.

**Comment by orbeckst** [comment]
Have you figured out how much work it is going to be to add h-bond analysis?

**Comment by ALescoulie** [comment]
I looked at the MDAnalysis docs and think it will not be too much work to implement a H-Bond analysis. I want to try making a decorator  that runs an AnalysisBase subclass i, It would simply require making a few generic methods for saving the results for each ensemble.

**Comment by ALescoulie** [comment]
I'll have a draft pull request for EnsembleWrapper up hopefully by tonight. I'm writing it as a decorator for AnalysisBase. It does add a new class, but it's fairly rudimentary, just passing data into the original Analysis.

**Comment by orbeckst** [comment]
@cadeduckworth here's the issue for adding H-bond analysis, in case you want to have a go.

Perhaps @ALescoulie could help with reviewing once you get a PR up?

**Comment by cadeduckworth** [comment]
Yes that sounds great! I am reading through the MDAnalysis docs now. There is an 'autocorrelation' module for hydrogen bond calculations that may also be of use.

**Comment by cadeduckworth** [comment]
After playing around with it, it appears that the most useful information can be obtained by specifying the solute of interest as an acceptor for one run, specifying the solute as a donor for another, generating data frames from each of these two (results)arrays, and then concatenating the data frames. So far it seems like this is the quickest way to get information only pertaining to H-bond accepting/donating by the solute. I think there are more complicated alternatives available regarding combined selection methods, but it does not seem like those methods would capture instances where the solute forms an H-bond with itself from some examples shown. However, using the split acceptor/donor method I explained above results in some overlap of the internal H-bonds, which can possibly be removed when concatenating the two data frames if sorted properly. I am going to try out a few more things and then upload the notebook to a private repository so y'all can review the input/output generated so far. From my examples it seems that it would not take much more work to implement ensemble analysis once we decide the best way to choose acceptor/donor selections.

**Comment by orbeckst** [comment]
You should not need to do two runs. If you define donors and acceptors appropriately then you should only need one run and then you can sort the resulting list of H-bonds appropriately. Generally, we really want to avoid to analyze trajectories too often and rather work with generated data.

**Comment by orbeckst** [comment]
Also HydrogenBondAnalysis has the `between=True` keyword to only look for inter-group H-bonds https://userguide.mdanalysis.org/stable/examples/analysis/hydrogen_bonds/hbonds-selections.html#Hydrogen-bonds-between-specific-groups

See also tutorial in https://userguide.mdanalysis.org/stable/examples/analysis/hydrogen_bonds/hbonds.html

**Comment by cadeduckworth** [comment]
@cadeduckworth 
-reminder for myself to create a pull request
